src/scitokens/utils/keycache.py

```diff
@@ -113,15 +119,34 @@ def getkeyinfo(self, issuer, key_id=None, insecure=False):
         conn.row_factory = sqlite3.Row
         curs = conn.cursor()
         curs.execute(key_query.format(issuer=issuer, key_id=key_id))
-        
+
         row = curs.fetchone()
         if row != None:
-            if self._check_validity(row):
-                # Convert the PEM formatted public key to a public key object
+            # If it's time to update the key, but the key is still valid
+            if int(row['next_update']) < time.time() and self._check_validity(row):
+                # Try to update the key, but if it doesn't work, just return the saved one
+                try:
+                    public_key, cache_timer = self._get_issuer_publickey(issuer, key_id, insecure)
```

Comment:
Should we hold the database lock across a network call?  Any reason to not drop it?